tests/actions/IOUTest.ts

```diff
@@ -0,0 +1,3197 @@
+                                        Onyx.disconnect(connectionID);
+
+                                        // A chat report, a transaction thread, and an iou report should be created
+                                        const chatReports = Object.values(allReports ?? {}).filter((report) => report?.type === CONST.REPORT.TYPE.CHAT);
+                                        const iouReports = Object.values(allReports ?? {}).filter((report) => report?.type === CONST.REPORT.TYPE.IOU);
+                                        expect(Object.keys(chatReports).length).toBe(2);
+                                        expect(Object.keys(iouReports).length).toBe(1);
+                                        const chatReport = chatReports[0];
+                                        const transactionThreadReport = chatReports[1];
+                                        const iouReport = iouReports[0];
+                                        iouReportID = iouReport?.reportID;
+                                        transactionThread = transactionThreadReport;
+
+                                        expect(iouReport?.notificationPreference).toBe(CONST.REPORT.NOTIFICATION_PREFERENCE.HIDDEN);
+
+                                        // They should be linked together
+                                        expect(chatReport?.participantAccountIDs).toEqual([CARLOS_ACCOUNT_ID]);
+                                        expect(chatReport?.iouReportID).toBe(iouReport?.reportID);
+
+                                        resolve();
+                                    },
+                                });
+                            }),
+                    )
+                    .then(
+                        () =>
+                            new Promise<void>((resolve) => {
+                                const connectionID = Onyx.connect({
+                                    key: `${ONYXKEYS.COLLECTION.REPORT_ACTIONS}${iouReportID}`,
+                                    waitForCollectionCallback: false,
+                                    callback: (reportActionsForIOUReport) => {
+                                        Onyx.disconnect(connectionID);
+
+                                        // The IOU report should have a CREATED action and IOU action
+                                        expect(Object.values(reportActionsForIOUReport ?? {}).length).toBe(2);
+                                        const createdActions = Object.values(reportActionsForIOUReport ?? {}).filter(
+                                            (reportAction) => (reportAction?.actionName as unknown as string) === CONST.REPORT.ACTIONS.TYPE.CREATED,
+                                        );
+                                        const iouActions = Object.values(reportActionsForIOUReport ?? {}).filter(
+                                            (reportAction) => (reportAction?.actionName as unknown as string) === CONST.REPORT.ACTIONS.TYPE.IOU,
```

Comment:
```suggestion
                                            (reportAction): reportAction is ReportActionBase & OriginalMessageIOU => reportAction?.actionName === CONST.REPORT.ACTIONS.TYPE.IOU,
```
This part `reportAction is ReportActionBase & OriginalMessageIOU` should help to get rid of `as IOUMessage` assertions below.